images/linux/scripts/installers/dotnetcore-sdk.sh

```diff
@@ -46,34 +46,32 @@ done
 sdks=()
 for version in ${versions[@]}; do
     release_url="https://dotnetcli.blob.core.windows.net/dotnet/release-metadata/${version}/releases.json"
-    echo "${release_url}"
-    releases=$(curl "${release_url}")
+    download_with_retries "${release_url}" "." "${version}.json"
+    releases=$(cat "./${version}.json")
     sdks=("${sdks[@]}" $(echo "${releases}" | jq '.releases[]' | jq '.sdk.version'))
     sdks=("${sdks[@]}" $(echo "${releases}" | jq '.releases[]' | jq '.sdks[]?' | jq '.version'))
+    rm ./${version}.json
 done
 
 sortedSdks=$(echo ${sdks[@]} | tr ' ' '\n' | grep -v preview | grep -v rc | grep -v display | cut -d\" -f2 | sort -u -r)
 
-for sdk in $sortedSdks; do
-    url="https://dotnetcli.blob.core.windows.net/dotnet/Sdk/$sdk/dotnet-sdk-$sdk-linux-x64.tar.gz"
-    echo "$url" >> urls
-    echo "Adding $url to list to download later"
-done
+# Download/install additional SDKs in parallel
+export -f download_with_retries
 
-# Download additional SDKs
-echo "Downloading release tarballs..."
-cat urls | xargs -n 1 -P 16 wget -q
-for tarball in *.tar.gz; do
-    dest="./tmp-$(basename -s .tar.gz $tarball)"
-    echo "Extracting $tarball to $dest"
-    mkdir "$dest" && tar -C "$dest" -xzf "$tarball"
-    rsync -qav "$dest/shared/" /usr/share/dotnet/shared/
-    rsync -qav "$dest/host/" /usr/share/dotnet/host/
-    rsync -qav "$dest/sdk/" /usr/share/dotnet/sdk/
-    rm -rf "$dest"
-    rm "$tarball"
-done
-rm urls
+parallel --jobs 0 --halt soon,fail=1 \
+    'url="https://dotnetcli.blob.core.windows.net/dotnet/Sdk/{}/dotnet-sdk-{}-linux-x64.tar.gz"; \
+    download_with_retries $url' ::: "${sortedSdks[@]}"
+
+find . -name "*.tar.gz" | parallel --halt soon,fail=1 \
+    'set -e
+    dest="./tmp-$(basename -s .tar.gz {})"; \
+    echo "Extracting {} to $dest"; \
+    mkdir "$dest" && tar -C "$dest" -xzf "{}"; \
+    rsync -qav "$dest/shared/" /usr/share/dotnet/shared/; \
```

Comment:
Can we move a folder instead of copy the content?

Reply:
Checked these folders, it seems that this stage need merge archive's content is different, so rsync is used, and it seems that rsync use is better than mv in this particular case

Reply:
This stage doesn't need to merge anything, just move folder A-> B. And a move operation in the same partition will be faster than copy, plus the `rm -rf "$dest" "$ARCHIVE_NAME"` will be faster due to delete less files.

Reply:
ok, it will be faster, true :)